src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,81 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.Ui;
+import duke.task.TaskList;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Command is used to represent a command from a user. All command has
+ * a execute method to execute the user's input.
+ */
+public abstract class Command {
+    protected String command;
+    protected String description;
+    protected String preposition;
+    protected LocalDate date;
+    protected boolean isExit;
+
+    Command(String command, String description, String preposition, LocalDate date, boolean isExit) {
+        this.command = command;
+        this.description = description;
+        this.preposition = preposition;
+        this.date = date;
+        this.isExit = isExit;
+    }
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+
+    public boolean isExit() {
+        return this.isExit;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        } else if (obj instanceof Command) {
+            Command other = (Command) obj;
+            boolean b1 = checkEquality(this.command, other.command);
+            boolean b2 = checkEquality(this.description, other.description);
+            boolean b3 = checkEquality(this.preposition, other.preposition);
+            boolean b4 = checkEqualityDate(this.date, other.date);
+            boolean b5 = this.isExit == other.isExit;
+
+            return b1 && b2 && b3 && b4 && b5;
+
+        } else {
+            return false;
+        }
+    }
+
+
+    private boolean checkEquality(String s1, String s2) {
+        if (s1 == null) {
+            return s2 == null;
+        } else {
+            if (s2 == null) {
+                return false;
+            } else {
+                return s1.equals(s2);
+            }
+        }
+    }
+
+    private boolean checkEqualityDate(LocalDate d1, LocalDate d2) {
```

Comment:
Maybe write it as checkDateEquality?
Though iirc LocalDate has an equals method already
https://beginnersbook.com/2017/10/java-localdate-equals/